scripts/west_commands/run_common.py

```diff
@@ -165,21 +194,113 @@ def do_run_common(command, user_args, user_runner_args, domains=None):
+
+        # Process all domains to load board names and populate flash runner
+        # parameters.
+        board_names = []
+        for d in domains:
+            if d.build_dir is None:
+                build_dir = get_build_dir(user_args)
+            else:
+                build_dir = d.build_dir
+
+            cache = load_cmake_cache(build_dir, user_args)
+            board = cache['CACHED_BOARD']
+
+            if board not in board_names:
+                board_names.append(board)
+
+            if board in board_image_count:
+                board_image_count[board].total = board_image_count[board].total + 1
+            else:
+                # This sets up the default entry for a board, no flashed images
+                # and one image in total.
+                board_image_count[board] = ImagesFlashed(0, 1)
+
+            # Load board flash runner configuration (if it exists) and store
+            # single-use commands in a dictionary so that they get executed
+            # once per unique board name.
+            if cache['BOARD_DIR'] not in processed_boards and 'SOC_FULL_DIR' in cache:
+                soc_yaml_file = Path(cache['SOC_FULL_DIR']) / 'soc.yml'
+                board_yaml_file = Path(cache['BOARD_DIR']) / 'board.yml'
+                group_type = 'boards'
+
+                # Search for flash runner configuration, board takes priority over SoC
+                try:
+                    with open(board_yaml_file, 'r') as f:
+                        data_yaml = yaml.safe_load(f.read())
+
+                except FileNotFoundError:
+                    continue
+
+                if 'runners' not in data_yaml:
+                    # Check SoC file
+                    group_type = 'qualifiers'
+                    try:
+                        with open(soc_yaml_file, 'r') as f:
+                            data_yaml = yaml.safe_load(f.read())
+
+                    except FileNotFoundError:
+                        continue
+
+                processed_boards[cache['BOARD_DIR']] = True
+
+                if 'runners' not in data_yaml or 'run_once' not in data_yaml['runners']:
+                    continue
+
+                for grp in data_yaml['runners']['run_once']:
+                    for cmd, data in grp.items():
+                        for group in data['groups']:
+                            run_first = bool(data['run'] == 'first')
+                            used_cmds.append(UsedFlashCommand(cmd, group[group_type], data['runners'], run_first))
+
+            if 'SOC_FULL_DIR' in cache  and len(used_cmds) > 0:
+                # Replace regex board names with full board names here so that commands can be
+                # tracked
+                for i, entry in enumerate(used_cmds):
+                    for l, match in enumerate(entry.boards):
+                        if match[0:1] == '/' and match[:1] == '/' and \
+                           re.match(match[1:len(match)-1], board) is not None:
```

Comment:
If you want to align the regex usage with snippets, then this should be a full match:
```suggestion
                           re.match(fr'^{match[1:-1]}$', board) is not None:
```

Reply:
Fixed